app/src/main/java/com/android/bookswap/ui/map/Map.kt

```diff
@@ -86,16 +89,23 @@ fun MapScreen(
     topAppBar: @Composable () -> Unit = {},
     bottomAppBar: @Composable () -> Unit = {},
 ) {
+  val context = LocalContext.current
+  var isOnline = remember { isNetworkAvailable(context) }
   val cameraPositionState = rememberCameraPositionState()
   // Get the user's current location
   val latitude = geolocation.latitude.collectAsState()
   val longitude = geolocation.longitude.collectAsState()
   // Start location and books updates
   LaunchedEffect(Unit) {
-    bookManagerViewModel.startUpdatingBooks()
-    geolocation.startLocationUpdates()
-    cameraPositionState.position =
-        CameraPosition.fromLatLngZoom(LatLng(latitude.value, longitude.value), INIT_ZOOM)
+    if (isOnline) {
+      bookManagerViewModel.startUpdatingBooks()
+      geolocation.startLocationUpdates()
+      cameraPositionState.position =
+          CameraPosition.fromLatLngZoom(LatLng(latitude.value, longitude.value), INIT_ZOOM)
+    } else {
+      Toast.makeText(context, "Please connect to Internet to actualise", Toast.LENGTH_SHORT).show()
```

Comment:
the correct word you are looking for in this toast is update, not actualise

Reply:
I would've shown either "There was a network error" or "Network has no internet access" for the text since these are strings already used in the Android API.
Both are found in Android API > res > values > strings.xml, with keys `httpError` and `other_networks_no_internet` and in the translated values using the same keys.

Using strings already used in Android lets us copy-paste translations rather than do the translation ourselves and also ensures proper string length for accessibility.

If needed, a dismissable snackbar can be used instead of a toast message to display additional info concerning the error, using a modal dialogue shown when the snackbar is clicked.